Python/preconfig.c

```diff
@@ -268,31 +268,22 @@ _PyPreCmdline_Read(_PyPreCmdline *cmdline, const PyPreConfig *preconfig)
 
 /* --- PyPreConfig ----------------------------------------------- */
 
-
-static PyStatus
-preconfig_check_struct_size(PyPreConfig *config)
-{
-    if (config->struct_size != sizeof(PyPreConfig)) {
-        return _PyStatus_ERR("unsupported PyPreConfig structure size "
-                             "(Python version mismatch?)");
-    }
-    return _PyStatus_OK();
-}
-
-
 PyStatus
 _PyPreConfig_InitCompatConfig(PyPreConfig *config)
 {
-    size_t struct_size = config->struct_size;
-    memset(config, 0, sizeof(*config));
-    config->struct_size = struct_size;
-
-    PyStatus status = preconfig_check_struct_size(config);
+    uint32_t header_version = config->header_version;
+    PyStatus status = _Py_CheckVersionCompat(header_version);
     if (_PyStatus_EXCEPTION(status)) {
         _PyStatus_UPDATE_FUNC(status);
         return status;
     }
 
+    // Future enhancement: allow old config struct layouts if an old
+    // header_version is specified
+    size_t struct_size = sizeof(*config);
+    memset(config, 0, struct_size);
+    config->header_version = header_version;
```

Comment:
As for PyConfig, the initial code didn't pay attention to the `struct_size` setting (it wasn't used in the `memset` call).